File: CP-Excercise/B_Tenzing_and_Books.py
def solve():
    n, x = map(int, input().split())
    a = list(map(int, input().split()))
    b = list(map(int, input().split()))
    c = list(map(int, input().split()))
    ans = 0

    for i in range(n):
        if a[i] > x:
            break
        temp = x
        kk = a[i]
        flag = True
        while temp > 0 and kk > 0:
            bit1 = temp & 1
            bit2 = kk & 1
            if bit1 == 0 and bit2 == 1:
                flag = False
                break
            temp >>= 1
            kk >>= 1
        if flag:
            ans |= a[i]
        else:
            break

    for i in range(n):
        if b[i] > x:
            break
        temp = x
        kk = b[i]
        flag = True
        while temp > 0 and kk > 0:
            bit1 = temp & 1
            bit2 = kk & 1
            if bit1 == 0 and bit2 == 1:
                flag = False
                break
            temp >>= 1
            kk >>= 1
        if flag:
            ans |= b[i]
        else:
            break

    for i in range(n):
        if c[i] > x:
            break
        temp = x
        kk = c[i]
        flag = True
        while temp > 0 and kk > 0:
            bit1 = temp & 1
            bit2 = kk & 1
            if bit1 == 0 and bit2 == 1:
                flag = False
                break
            temp >>= 1
            kk >>= 1
        if flag:
            ans |= c[i]
        else:
            break

    if ans == x:
        print("Yes")
    else:
        print("No")


t = int(input())
for _ in range(t):
    solve()

File: CP-Excercise/B_Tenzing_and_Books.cpp
#include<bits/stdc++.h>
using namespace std;

int main() {
    int t; cin>>t;
    while(t--){
        int n,x; 
        cin>>n>>x;
        int stack1[n+5] = {0},stack2[n+5] = {0},stack3[n+5] = {0};
        for (int i=0;i<n;i++) cin>>stack1[i];
        for (int i=0;i<n;i++) cin>>stack2[i]; 
        for (int i=0;i<n;i++) cin>>stack3[i]; 
        int ans=0;
        for (int i=0;i<n;i++) 
        {
            if(stack1[i]>x) break;
            int store=x;
            int stacking =stack1[i];
            bool checker=true;
            while(store>0 and stacking>0) {
                int bit1 = store&1;
                int bit2 =stacking&1;
                if (bit1==0 and bit2==1) {checker=false; break; }
                store>>=1;
                stacking>>=1;
                }
            if(checker) ans|=stack1[i];
            else break;
        }
        for (int i=0;i<n;i++) 
        {
            if(stack2[i]>x) break;
            int store=x;
            int stacking =stack2[i];
            bool checker=true;
            while(store>0 and stacking>0) {
                int bit1 = store&1;
                int bit2 =stacking&1;
                if (bit1==0 and bit2==1) 
                {
                    checker=false;
                    break;
                }
                store>>=1;
                stacking>>=1;
                }
            if(checker) ans|=stack2[i];
            else break;
        }
        for (int i=0;i<n;i++) 
        {
            if(stack3[i]>x) break;
            int store=x;
            int stacking =stack3[i];
            bool checker=true;
            while(store>0 and stacking>0) {
                int bit1 = store&1;
                int bit2 =stacking&1;
                if (bit1==0 and bit2==1) 
                {
                    checker=false;
                    break;
                }
                store>>=1;
                stacking>>=1;
                }
            if(checker) ans|=stack3[i];
            else break;
        }
        if(ans==x) cout<<"Yes\n";
        else cout<<"No\n";
    }
    return 0;
}  